android/src/main/java/com/oney/WebRTCModule/PermissionUtils.java

```diff
@@ -0,0 +1,133 @@
+     */
+    private static final String GRANT_RESULT = "GRANT_RESULT";
+    private static final String PERMISSION = "PERMISSION";
+    private static final String REQUEST_CODE = "REQUEST_CODE";
+    private static final String RESULT_RECEIVER = "RESULT_RECEIVER";
+
+    /**
+     * Incrementing counter for permissions requests. Each request must have a unique numeric code.
+     */
+    private static int requestCode;
+
+    /**
+     * Requests the given permission. The callback will we called with the requested permission and
+     * the granted result. See: https://developer.android.com/reference/android/content/pm/PackageManager.html#PERMISSION_GRANTED
+     * for the result constants.
+     *
+     * @param context Application context / activity.
+     * @param permission Permission which the application is interested in requesting.
+     * @param callback Callback where the results will be reported.
+     */
+    public static void requestPermission(Context context,
+                                         String permission,
+                                         final Callback callback) {
+        ResultReceiver resultReceiver = new ResultReceiver(new Handler(Looper.getMainLooper())) {
+            @Override
+            protected void onReceiveResult (int resultCode, Bundle resultData) {
+                String permission = resultData.getString(PERMISSION);
+                int grantResult = resultData.getInt(GRANT_RESULT);
+
+                callback.invoke(permission, grantResult);
+            }
+        };
+
+        requestCode++;
+
+        Intent permissionsIntent = new Intent(context, PermissionHelperActivity.class);
+        permissionsIntent.putExtra(RESULT_RECEIVER, resultReceiver);
+        permissionsIntent.putExtra(PERMISSION, permission);
+        permissionsIntent.putExtra(REQUEST_CODE, requestCode);
+        permissionsIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        context.startActivity(permissionsIntent);
+    }
+
+    /**
+     * Helper activity for requesting permissions. Android only allows requesting permissions from
+     * an activity and the result is reported in the <tt>onRequestPermissionsResult</tt> method.
+     * Since this package is a library we create an auxiliary activity and communicate back the
+     * results using a <tt>ResultReceiver</tt>.
+     */
+    public static class PermissionHelperActivity extends Activity {
+        int requestCode;
+        ResultReceiver resultReceiver;
+        String permission;
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onRequestPermissionsResult(int requestCode,
+                                               String permissions[],
+                                               int[] grantResults) {
+            Bundle resultData = new Bundle();
+            resultData.putString(PERMISSION, permissions[0]);
+            resultData.putInt(GRANT_RESULT, grantResults[0]);
```

Comment:
A few observations of mine about this PR will follow. I've taken care of them and I'm putting them here in writing for the sake of clarity and completeness.

The super's documentation says: It is possible that the permissions request interaction with the user is interrupted. In this case you will receive empty permissions and results arrays which should be treated as a cancellation.

Hence, the above seem opportunistic to me.